src/Controllers/FoodVoucherController.php

```diff
@@ -0,0 +1,238 @@
+use Engelsystem\Models\Worklog;
+use ErrorException;
+use GuzzleHttp\Client as GuzzleClient;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Log\LoggerInterface;
+
+class FoodVoucherController extends BaseController
+{
+    use HasUserNotifications;
+
+    public function __construct(
+        protected Authenticator $auth,
+        protected GuzzleClient $guzzle,
+        protected Config $config,
+        protected LoggerInterface $log,
+        protected Worklog $worklog,
+        protected Redirector $redirect,
+        protected Response $response,
+        protected User $user,
+        protected Translator $translator,
+    ) {
+    }
+
+    private function checkActive(): void
+    {
+        if (!$this->config->get('enable_voucher') || !$this->config->get('enable_force_food')) {
+            throw new HttpNotFound();
+        }
+        if (!$this->userIsForced() && UserVouchers::eligibleVoucherCount($this->auth->user()) == 0) {
+            throw new HttpForbidden();
+        }
+    }
+
+    private function checkConfig(): void
+    {
+        $conf = $this->config->get('food_voucher_api');
+        if (
+            !(
+            $conf
+            && $conf['info_url']
+            && $conf['auth_token']
+            && $conf['post_url']
+            )
+        ) {
+            throw new HttpNotFound();
+        }
+    }
+
+    private function getAuthToken(): string
+    {
+        return $this->config->get('food_voucher_api')['auth_token'];
+    }
+
+    private function userIsForced(): bool
+    {
+        $user = $this->auth->user();
+        return $user->state->force_active && $this->config->get('enable_force_active')
+            || $user->state->force_food && $this->config->get('enable_force_food');
+    }
+
+    /**
+     * @throws ErrorException
+     */
+    private function getInfo(bool $crew, ?array $userMealVouchers = null): array
+    {
+        $this->checkConfig();
+
+        $data = cache('foodVoucherInfo', function () {
+            $infoUrl = (string) $this->config->get('food_voucher_api')['info_url'];
+            try {
+                $response = $this->guzzle->get(
+                    $infoUrl,
+                    ['headers' =>
+                        [
+                            'Authorization' => 'Bearer ' . $this->getAuthToken(),
+                            'Content-Type' => 'application/json',
+                        ],
+                    ]
+                );
+            } catch (ConnectException | GuzzleException $e) {
+                $this->log->error('Exception during food voucher api request', ['exception' => $e]);
+                throw new ErrorException('user.food.request-error');
+            }
+            if ($response->getStatusCode() !== 200) {
+                throw new HttpNotFound('user.food.request-error');
+            }
+            return json_decode($response->getBody()->getContents(), true);
+        }, 60);
```

Comment:
Making the cache time optionally configurable would spare us "live patching" and/or "fast tracking deployments" during the event which i would prefer